operator/pkg/common/extensions/secret_reconciliator.go

```diff
@@ -38,58 +37,57 @@ func (r *SecretReconciliator) Client() ctrlClient.Client {
 	return r.client
 }
 
-// Namespace returns the namespace of the object the secret is owned by
-func (r *SecretReconciliator) Namespace() string {
-	return r.obj.GetNamespace()
+// DeleteSecret makes sure a secret with the given name does NOT exist.
+// NOTE that this function will never touch a secret which is not owned by the object passed to the constructor.
+func (r *SecretReconciliator) DeleteSecret(ctx context.Context, name string) error {
+	secret := &coreV1.Secret{}
+	key := ctrlClient.ObjectKey{Namespace: r.obj.GetNamespace(), Name: name}
+	if err := r.Client().Get(ctx, key, secret); err != nil {
+		if !apiErrors.IsNotFound(err) {
+			return errors.Wrapf(err, "checking existence of %s secret", name)
+		}
+		secret = nil
+	}
+	if secret == nil || !metav1.IsControlledBy(secret, r.obj) {
+		return nil
+	}
+
+	if err := utils.DeleteExact(ctx, r.Client(), secret); err != nil && !apiErrors.IsNotFound(err) {
+		return errors.Wrapf(err, "deleting %s secret", name)
+	}
+	return nil
 }
 
-// ReconcileSecret reconciles a secret with the given name, by making sure its existence matches "shouldExist" value.
+// ReconcileSecret reconciles a secret with the given name, by making sure it exists.
```

Comment:
Perhaps this should now be called `EnsureSecret`, WDYT @ebensh ?

Reply:
OK, will rename. The semantics discussed offline (a followup PR will remove the weird boolean).